app/controllers/shelters_controller.rb

```diff
@@ -12,14 +12,29 @@ def new
     end
 
     def create
-        shelter = Shelter.new({
-        name: params[:shelter][:name],
-        address: params[:shelter][:address],
-        city: params[:shelter][:city],
-        state: params[:shelter][:state],
-        zip: params[:shelter][:zip]
-        })
+        shelter = Shelter.create(shelter_params)
+        redirect_to "/shelters"
+    end
+
+    def edit
+        @shelter = Shelter.find(params[:id])
+    end
+
+    def update
+        shelter = Shelter.find(params[:id])
+        shelter.update(shelter_params)
         shelter.save
+        redirect_to "/shelters/#{shelter.id}"
+    end
+
+    def destroy
+        Shelter.destroy(params[:id])
         redirect_to "/shelters"
```

Comment:
this now destroys the shelter record and sends a user back to the shelters index page.